src/flyout.js

```diff
@@ -195,6 +195,37 @@ export class FlyoutElement extends LitElement {
     `;
   }
 
+  _getFlyoutLinkWithFilename = (url) => {
+    // Get the resolver's filename returned by the application (as HTTP header)
+    // and injected by Cloudflare Worker as a meta HTML tag
+    const metaFilename = document.querySelector(
+      "meta[name='readthedocs-resolver-filename']",
+    );
+
+    // Remove trailing slashes from the version's URL and append the
+    // resolver's filename after removing trailing ``index.html``.
+    // Examples:
+    //
+    //   URL: https://docs.readthedocs.io/en/latest/
+    //   Filename: /index.html
+    //   Flyuout URL: https://docs.readthedocs.io/en/latest/
+    //
+    //   URL: https://docs.readthedocs.io/en/stable/
+    //   Filename: /guides/access/index.html
+    //   Flyuout URL: https://docs.readthedocs.io/en/stable/guides/access/
+
+    // Keep only one trailing slash
+    const base = url.replace(/\/+$/, "/");
+
+    // 1. remove initial slash to make it relative to the base
+    // 2. remove the trailing "index.html"
+    const filename = metaFilename.content
+      .replace(/^\//, "")
+      .replace(/index.html$/, "");
```

Comment:
Otherwise this may introduce some bugs for paths that end with index.html `foo/test-index.html`

```suggestion
      .replace(/\/index.html$/, "/");
```

But should probably come before the first replace, or a path like `/index.html` won't work.

Reply:
Hrm, good point. I reverted this change because `.replace(/\/index.html$/, "/")` doesn't work as expected in the local tests. I will take another look at this.

Reply:
Thanks for the review on this. I fixed it on b5b23b3